### PR TITLE
Add 'equals' group constraints; supply -> carrier_prod

### PR DESCRIPTION
**Pull request by brynpickering**
Summary of changes in this pull request:

* Added 'equals' constraints for a few existing group constraints 
* Changed 'supply' to 'carrier_prod' for a recently added group constraint (although I'm now not sure this was a good idea, since it no longer matches e.g. 'supply_share' group constraints)

Reviewer checklist:

- [x] Test(s) added to cover contribution
- [x] Documentation updated
- [x] Changelog updated
- [x] Coverage maintained or improved

### CODE REVIEW

File: changelog.rst [hunk 1]
```diff
@@ -6,7 +6,9 @@ Release History
 0.6.5 (dev)
 -----------
 
-|new| New model-wide constraints `supply_min` and `supply_max` which restrict the absolute energy produced by a subgroup of technologies and locations.
+|new| New model-wide constraints `energy_cap_equals`, `resource_area_equals`, and  `energy_cap_share_equals` to add the equality constraint to existing `min/max` group constraints.
+
+|new| New model-wide constraints `carrier_prod_min`, `carrier_prod_max`, and  `carrier_prod_equals` which restrict the absolute energy produced by a subgroup of technologies and locations.
```
Comment (sjpfenninger) [inline changelog.rst]: We should at least mention that this is part new and part renaming of existing constraints

File: calliope/config/defaults.yaml [hunk 2]
```diff
@@ -89,7 +89,7 @@ tech_groups:
...
+        allowed_group_constraints: [demand_share_min, demand_share_max, demand_share_equals, demand_share_per_timestep_min, demand_share_per_timestep_max, demand_share_per_timestep_equals, demand_share_per_timestep_decision, supply_share_min, supply_share_max, supply_share_equals, supply_share_per_timestep_min, supply_share_per_timestep_max, supply_share_per_timestep_equals, cost_max, cost_min, cost_equals, cost_var_max, cost_var_min, cost_var_equals, cost_investment_min, cost_investment_max, cost_investment_equals, energy_cap_share_min, energy_cap_share_max, energy_cap_share_equals, energy_cap_min, energy_cap_max, energy_cap_equals, resource_area_min, resource_area_max, resource_area_equals, carrier_prod_min, carrier_prod_max, carrier_prod_equals]
```
Comment (sjpfenninger) [inline calliope/config/defaults.yaml]: We should rename all `demand_share` to `carrier_con_share` and `supply_share` to `carrier_prod_share` for consistency

Comment (brynpickering) [reply calliope/config/defaults.yaml]: I don't think carrier_con_share makes sense, sadly, as we only consider demand, not other energy consuming techs (storage, conversion, and transmission)...

File: changelog.rst (hunk 1 above)
Comment (brynpickering) [reply changelog.rst]: I think this is all new, as far 0.6.5 is concerned. They were introduced as `supply_max/min` in 0.6.5-dev, so renaming them doesn't really matter for the changelog, so long as the name is fixed by the next stable release?

**Pull request merged**